MethodToMethodWithCheckRector to replace methods with a certainty check

bbrala opened the pull request:
## Description
MethodToMethodWithCheckRector to replace methods with a certainty check.

## To Test
- Add steps to test this feature

## Drupal.org issue
Provide a link to the issue from https://www.drupal.org/project/rector/issues. If no issue exists, please create one and link to this PR.

bbrala committed:
MethodToMethodWithCheckRector to replace methods with a certainty check

bbrala committed:
Simplify FQCN

bbrala committed:
Move class and do something weird to the comment.

bbrala committed:
simplify fqcn

mglaman commented on src/Rector/Deprecation/MethodToMethodWithCheckRector.php:
```diff
@@ -0,0 +1,187 @@
+            $node_var = $node->var->name;
+
+            if ($node->var instanceof Node\Expr\Variable) {
+                $node_var = "$$node_var";
```
Do you know if something needs to be escaped here? Because we're doing that weird PHP thing with double `$$`.

Would this be `$node_var = "${$node_var}";`

bbrala replied:
It is kinda weird to do something like this while we should have the power of rector and phpstan. 

Perhaps I should just try and add the failing case to a unit test, if that also fails debugging is a little easier.

bbrala replied:
Well, it's only for th printing, and it seems to do nothing special.

bbrala committed:
Fix broken expectation for d8 fixture.

It seems the analysis got better in a way Changing to logical expected class name. and added to unit tests.

bbrala committed:
It seems rector is very sure of the type of this specific example.

bbrala committed:
Extra whitespace that was removed

bbrala committed:
Merge branch 'main' into feature/method-to-method-with-check-rector

mglaman committed:
Merge branch 'main' into feature/method-to-method-with-check-rector

mglaman reviewed (dismissed):
phpstan errors :/

bbrala committed:
Make code more readable by phpstan by making it less readable for humans. Thanks phpstan.

bbrala committed:
Remove removed class from baseline.

agentrickard committed:
Merge branch 'main' into feature/method-to-method-with-check-rector
